doc/CONFIGURATION.md

```diff
@@ -141,3 +149,453 @@ node:
+
+# TemporalX
+
+The `temporalx` section is used to configure the TemporalX gRPC server that facilitates interaction with our custom node. Additionally it enables things like tracing, and monitoring.
+
+<p align="left">
+  · <a href="#api"><strong>API</strong></a>
+  <br>
+  · <a href="#prometheus"><strong>Prometheus</strong></a>
+  <br>
+  · <a href="#profiling"><strong>Profiling</strong></a>
+</p>
+
+## API
+
+The `api` section is used to configure the gRPC API server. 
+
+Configuration Options:
+
+* `tracing` enables/disables jaeger tracing
+* `listen_address` is the address to listen on for gRPC connections
+* `listen_proto` is the protocol the gRPC endpoint is using
+* `tls.cert_file` is the location to a tls certificate
+* `tls.key_file` is the location to the tls certificate key file
+* `jwt.realm` is the realm of the jwt
+* `jwt.key` is they used for signing jwt's
+
+## Prometheus
+
+The `prometheus` section is used to configure a prometheus endpoint that can be used for metric collection. The default setting is disabled. Metric collection via prometheus can sometimes be impactful on resources
+
+Configuration Options:
+
+* `enabled` enables/disables the prometheus endpoint
+* `endpoint` the ip+port to expose the `/metrics` handler on.
+
+## Profiling
+
+The `profiling` section is used to configure profiling of TemporalX via pprof, and exposes `net/http/pprof` handlers.
+
+Configuration Options:
+
+* `enabled` enables/disables profiling collection
+* `endpoint` is the ip+port to expose `net/http/pprof` handlers on.
+
+# Node Configuration
+
+The `node` section is used to configure the underlying libp2p, and ipfs subsystems used by TemporalX.
+
+<p align="left">
+  · <a href="#storage"><strong>Storage</strong></a>
+  <br>
+  · <a href="#peerstore"><strong>Peerstore</strong></a>
+  <br>
+  · <a href="#keystore"><strong>Keystore</strong></a>
+  <br>
+  · <a href="#libp2p"><strong>LibP2P</strong></a> 
+  <br>
+  · <a href="#opts"><strong>Opts</strong></a> 
+</p>
+
+## Storage
+
+The `storage` section is used to configure the main storage layer of our node. It consists of a generic `datastore` with a `blockstore` ontop, and is primarily used for storing our "data" (files, etc..). It also enables managing of blocks using a novel reference count system, as opposed to a pinning system.
+
+This is also where we configure the reference counted blockstore, which is an optional feature that is disabled by default. To read about the reference counter, click [here](REFERENCE_COUNTER.md).
+
+When not using a reference counted blockstore, delete requests are not blocked, even if you are attempting to delete a block that may be referenced/used by more than one other block. If you do this you can inadvertently lose data. Therefore when not using a reference counted blockstore, you must implement a data management system outside of TemporalX and ensure you consult with it before deleting any blocks.
```

Comment:
When not using a reference-counted blockstore, delete requests operate on the backing data directly. Therefore, an external data management system is required if the reference-counted blockstore is not enabled.

Reply:
good change